fedimint-api/src/db/mod.rs

```diff
@@ -169,10 +175,86 @@ impl Drop for CommitTracker {
     }
 }
 
+struct IsolatedDatabaseTransaction<'isolated, 'parent: 'isolated> {
+    tx: &'isolated mut DatabaseTransaction<'parent>,
+    prefix: Vec<u8>,
+}
+
+impl<'isolated, 'parent: 'isolated> IsolatedDatabaseTransaction<'isolated, 'parent> {
+    pub fn new(
```

Comment:
In the future we might want to inject prefixes for other stuff, so something more generic than `ModuleInstanceId` might be needed, but good enough for now, and simpler. Just mentioning it - this struct could be used to "isolated by prefix" things other than modules just fine.

Reply:
I took a swing at this. IsolatedDatabaseTransaction now accepts anything that is Send + Encodable.